test/unit/flow_presenter_test.rb

```diff
@@ -164,4 +164,35 @@ def flow_registry
       assert_equal "/flow-name/s", flow_presenter.start_page_link
     end
   end
+
+  context "#back_link" do
+    should "return the start page on first question in smart flow" do
+      flow_presenter = FlowPresenter.new({}, @flow)
+      assert_equal("/flow-name", flow_presenter.back_link(0))
+    end
+
+    should "return the start page on first question in session flow" do
+      @flow.use_session(true)
+      params = { id: @flow.name, node_name: "node-name", responses: { "hello" => "there" } }
+      flow_presenter = FlowPresenter.new(params, @flow)
+      assert_equal("/flow-name", flow_presenter.back_link(0))
+    end
+
+    should "return the previous page for another question in smart flow" do
+      name = "calculate-your-holiday-entitlement"
+      flow = flow_registry.find(name)
+      params = { responses: "days-worked-per-week/starting", id: name }
+      flow_presenter = FlowPresenter.new(params, flow)
+      assert_equal("/#{name}/y/days-worked-per-week/starting", flow_presenter.back_link(3))
+    end
+
+    should "return the previous page for another question in session flow" do
+      name = "find-coronavirus-support"
+      flow = flow_registry.find(name)
+      params = { id: name, node_name: "feeling_unsafe", responses: { "need_help_with" => %w[feeling_unsafe] } }
+      flow_presenter = FlowPresenter.new(params, flow)
+
+      assert_equal("/#{name}/s/need-help-with", flow_presenter.back_link(2))
+    end
```

Comment:
These 2 tests seem a bit tricky to follow as you need to know about the flow to understand the test. Could they be done without using the flow_registry? (they seem to be the only 2 tests using it anyway so it could also be removed) 

I found myself looking for the @flow.use_session(true) of the earlier examples.

Reply:
I tried and failed to change these... will try again next time but may need some help.

Reply:
Thanks for helping me write these :) 🙏  I updated the commit.

Reply:
Super, they look great now 👍